source/agora/common/EnrollmentManager.d

```diff
@@ -125,6 +136,16 @@ public class EnrollmentManager
     {
         static ubyte[] buffer;
 
+        // check validity of the enrollment data
+        auto reason = isInvalidEnrollmentReason(this.ledger.getBlockHeight() + 1,
+            enroll, this.utxo_set.getUTXOFinder());
+        if (reason !is null)
```

Comment:
One nice idiom in D is the ability to do:
```D
if (auto reason = func())
```

Which looks like a prime feat for this.